### PR TITLE
C# code samples for Advanced Vector Math

### PR DESCRIPTION
**Pull request by KellyThomas**

### CODE REVIEW

**Comment by mhilbrunner** [comment]
cc @mysticfall

File: tutorials/math/vectors_advanced.rst [hunk 1]
```diff
@@ -242,6 +315,61 @@ Code should be something like this:
     if (overlapping):
         print("Polygons Collided!")
 
+ .. code-tab:: csharp
+
+        var overlapping = true;
+
+        foreach (Plane p in planes_of_A)
```
[the remark this answers is not in the document]
Comment (paulloz) [reply tutorials/math/vectors_advanced.rst]: My two cents: I'm not sure we should use LINQ in documentation examples. Or maybe just add a comment mentioning how LINQ could help here?

[the remark this answers is not in the document]
Comment (KellyThomas) [reply tutorials/math/vectors_advanced.rst]: These sections could IMO be simplified in both languages by restructuring to function with early returns.  But on the other hand I am reluctant to change the flow while merely translating.

[the remark this answers is not in the document]
Comment (mhilbrunner) [reply tutorials/math/vectors_advanced.rst]: IMO best just change the snake case names and *maybe* add a note about LINQ. If you want, you could create a separate PR to simplify the structure in both languages afterwards :)

[the remark this answers is not in the document]
Comment (mysticfall) [reply tutorials/math/vectors_advanced.rst]: Yeah, the mention about LINQ was more of a side note, so I agree with what @mhilbrunner said.

**Comment by mysticfall** [comment]
LGTM. Thanks much! :)

**Comment by KellyThomas** [comment]
Thanks everyone for helping looking over this.

Always good to have another set of eyes!

**Pull request merged**